### ISSUE TITLE
Use Gyroscope when available

### ISSUE THREAD

**Issue by GoogleCodeExporter** [sky-map-team/stardroid#5]
```
This is more of an enhancement request :) It would be nice if somebody could 
work on improving tracking by using the gyroscope in devices where it's 
available or a more complicated predictive filter (Kalman?)


```

Original issue reported on code.google.com by `nkatza...@gmail.com` on 24 Jan 2012 at 4:40

**Comment by barbeau** [comment]
FWIW, here's what I consider to be a fairly solid implementation of the ROTATION_VECTOR sensor licensed under Apache 2.0 - full disclosure, I implemented it.

The onSensorChanged() method:
https://github.com/barbeau/gpstest/blob/master/GPSTest/src/main/java/com/android/gpstest/GpsTestActivity.java#L561

Here's the initial setup in onResume():
https://github.com/barbeau/gpstest/blob/master/GPSTest/src/main/java/com/android/gpstest/GpsTestActivity.java#L229

...and teardown in onPause():
https://github.com/barbeau/gpstest/blob/master/GPSTest/src/main/java/com/android/gpstest/GpsTestActivity.java#L264

It supports:
* Back to API level 8 (Froyo) - uses `Sensor.TYPE_ORIENTATION` on devices that don't support `Sensor.TYPE_ROTATION_VECTOR`
* Orientation changes - remaps the coordinate system accordingly
* Correction for magnetic north
* A workaround for a Samsung-specific bug that causes a crash on certain Samsung devices (see  https://github.com/barbeau/gpstest/issues/39 for details)

**Comment by jaydeetay** [comment]
Implemented in the latest version (1.9.0b).
